pkg/gardenlet/operation/botanist/infrastructure.go

```diff
@@ -70,22 +72,28 @@ func (b *Botanist) WaitForInfrastructure(ctx context.Context) error {
 			return err
 		}
 
-		if len(b.Shoot.Components.Extensions.Infrastructure.ServicesCIDRs()) > 0 && len(b.Shoot.Components.Extensions.Infrastructure.PodsCIDRs()) > 0 {
-			if err := b.Shoot.UpdateInfoStatus(ctx, b.GardenClient, true, func(shoot *gardencorev1beta1.Shoot) error {
-				shoot.Status.Networking = &gardencorev1beta1.NetworkingStatus{
-					Nodes:    nodesCIDRs,
-					Services: b.Shoot.Components.Extensions.Infrastructure.ServicesCIDRs(),
-					Pods:     b.Shoot.Components.Extensions.Infrastructure.PodsCIDRs(),
-				}
-				return nil
-			}); err != nil {
-				return err
-			}
-		}
+		networkingStatus.Nodes = nodesCIDRs
+	}
 
-		if err := extensions.SyncClusterResourceToSeed(ctx, b.SeedClientSet.Client(), b.Shoot.SeedNamespace, b.Shoot.GetInfo(), nil, nil); err != nil {
-			return err
-		}
+	if servicesCIDRs := b.Shoot.Components.Extensions.Infrastructure.ServicesCIDRs(); len(servicesCIDRs) > 0 {
+		networkingStatus.Services = servicesCIDRs
+	}
+	if podsCIDRs := b.Shoot.Components.Extensions.Infrastructure.PodsCIDRs(); len(podsCIDRs) > 0 {
+		networkingStatus.Pods = podsCIDRs
+	}
+	if egressCIDRs := b.Shoot.Components.Extensions.Infrastructure.EgressCIDRs(); len(egressCIDRs) > 0 {
+		networkingStatus.Egress = egressCIDRs
+	}
+
+	if err := b.Shoot.UpdateInfoStatus(ctx, b.GardenClient, true, func(shoot *gardencorev1beta1.Shoot) error {
+		shoot.Status.Networking = networkingStatus
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if err := extensions.SyncClusterResourceToSeed(ctx, b.SeedClientSet.Client(), b.Shoot.SeedNamespace, b.Shoot.GetInfo(), nil, nil); err != nil {
+		return err
```

Comment:
This completely changes the flow in the sense that this is now the default case, i.e. it is always executed regardless if an extension set in the infrastructure status or not. I am not sure why this should pose a problem, but it certainly is a major change compared to the previous approach. Should we add some checks if `networkStatus` is still empty after all `if` blocks before writing it back?
WDYT?

Reply:
My thoughts were the same.
I eventually decided against checking for an empty `networkingStatus`, as it could lead to orphaning the `shoot.status.networking` section if the corresponding fields in the `Infrastructure` are no longer reported.
I think we should always mirror the `Infrastructure` status to prevent surprises and cover edge cases. Do you agree?